modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5040,6 +5048,16 @@ private int extrasSize() {
         lock.lock();
     }
 
+    /** {@inheritDoc} */
+    @Override public boolean tryLockEntry(long timeout) {
+        try {
+            return lock.tryLock(timeout, TimeUnit.MILLISECONDS);
+        }
+        catch (InterruptedException ignite) {
+            return false;
```

Comment:
We should return interrupt flag to current thread by calling Thread.currentThread().interrupt();
Probably, another piece of code will handle it.